pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/invertedindex/RealtimeLuceneTextIndexReader.java

```diff
@@ -89,6 +90,11 @@ public void add(String document) {
     _indexCreator.add(document);
   }
 
+  @Override
+  public ImmutableRoaringBitmap getDictIds(String searchQuery) {
+    throw new UnsupportedOperationException();
```

Comment:
Why do we need this method?

Reply:
Yeah I agree this is not ideal, but since FST index stores key -> dictid mapping I created this overridable method. 
I moved the FSTIndexCreator to extend TextReader interface too since that looks more closer than inverted index. 
Maybe this could be moved to it's own interface if there are similar indicies in future.